app/plugins/Files/js/buttons.js

```diff
@@ -127,6 +129,43 @@ eID('cancel-ascii').onclick = function() {
 	hide('show-ascii');
 };
 
+// Select directory sliding frame
+eID('new-dir').onclick = function() {
+	show('add-dir');
+	hide('file-library');
+};
+eID('back-dir').onclick = exitFileAdder;
+
+// Upload directory option chosen
+eID('upload-dir-choice').onclick = function() {
+	var path = require("path");
```

Comment:
Do not require libraries within commonly occurring events. I forget exactly how, but this causes a mem leak. I took to the convention of requiring libraries at the top of `helpers.js`